drivers/sensor/vl53l1x/vl53l1_platform.c

```diff
@@ -0,0 +1,197 @@
+/* vl53l1_platform.c - Zephyr customization of ST vl53l1x library.
+ */
+
+/*
+ * Copyright (c) 2023 Prosaris Solutions Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "vl53l1_platform.h"
+#include "vl53l1_platform_log.h"
+
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/init.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_DECLARE(VL53L1X, CONFIG_SENSOR_LOG_LEVEL);
+
+VL53L1_Error VL53L1_WriteMulti(VL53L1_Dev_t *pdev, uint16_t reg,
+		uint8_t *pdata, uint32_t count)
+{
+	VL53L1_Error status = VL53L1_ERROR_NONE;
+	int32_t status_int = 0;
+	uint8_t buffer[count + 2];
+
+	/* To be able to write to the 16-bit registers/addresses on the vl53l1x */
+	buffer[1] = (uint8_t)(reg & 0x00ff);
+	buffer[0] = (uint8_t)((reg & 0xff00) >> 8);
+
+	memcpy(&buffer[2], pdata, count);
+
+	status_int = i2c_write(pdev->i2c, buffer, count + 2, pdev->i2c_slave_address);
```

Comment:
Doing the following results in failure when `VL53L1_WriteMulti` when called by STM's library (tested on hardware). Replacing all other manual endian conversion with `sys_cpu_to_be16` (and the like) seems OK. I appreciate the suggestion, but I am going to leave this one.

```
VL53L1_Error VL53L1_WriteMulti(VL53L1_Dev_t *pdev, uint16_t reg,
		uint8_t *pdata, uint32_t count)
{
	VL53L1_Error status = VL53L1_ERROR_NONE;
	int32_t status_int = 0;

        reg = sys_cpu_to_be16(reg);

        status_int = i2c_write(pdev->i2c, (unit8_t *)&reg, 2, pdev->i2c_slave_address);
	status_int += i2c_write(pdev->i2c, pdata, count, pdev->i2c_slave_address);

	if (status_int < 0) {
		status = VL53L1_ERROR_CONTROL_INTERFACE;
		LOG_ERR("Failed to write");
	}

	return status;
}
```

Reply:
It won't work to use two write calls like that.  At least, it depends on the I2C device.  You'll get a two start, address, data, stop, transactions on I2C, which is usually not considered the same thing.  To do it in one transaction, it would be like i2c_burst_write() that uses i2c_transfer() with two messages.

But there's a note that not every i2c controller can do this and combining the data into one buffer is more compatible.  So, maybe you want to keep the bounce buffer like you've got.

But Zephyr still has a function to do the endianness for you.
```
sys_put_be16(reg, buffer);
```